sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -302,6 +303,24 @@ def test_lasso_cv_positive_constraint():
     assert min(clf_constrained.coef_) >= 0
 
 
+def _scale_alpha(estimator, n_samples):
+    if 'alpha' not in estimator.get_params():
+        return
+
+    model_name = estimator.__class__.__name__
+    if model_name in ['Lasso', 'LassoLars', 'MultiTaskLasso']:
+        alpha = estimator.alpha * np.sqrt(n_samples)
+    if model_name in ['Ridge', 'RidgeClassifier']:
+        alpha = estimator.alpha * n_samples
+    if model_name in ['ElasticNet', 'MultiTaskElasticNet']:
+        if estimator.l1_ratio == 1:
+            alpha = estimator.alpha * np.sqrt(n_samples)
+        if estimator.l1_ratio == 0:
+            alpha = estimator.alpha * n_samples
+
+    estimator.set_params(alpha=alpha)
```

Comment:
Since this is happening inplace, we could add it explicitly in the name of the function. To be honest, I would prefer to return a clone I think:


```
return clone(estimator).set_params(alpha=alpha)
```
```

Reply:
hm... then we would need to define the estimator, update it in the _scale_alpha and then put it in the pipeline? Slightly more complicated. I would prefer to leave it as is unless you see larger advantages over this option?

Reply:
I agree with @glemaitre's point: I think renaming the function would make the pipeline definition easier to follow:

```python
    model = estimator(normalize=False, fit_intercept=True, **params)
    model = _scale_alpha(model)
    reg_with_scaler = make_pipeline(
        StandardScaler(with_mean=with_mean),
        model,
    )
```

or alternatively:

```python
    model = estimator(normalize=False, fit_intercept=True, **params)
    _inplace_scale_alpha(model)
    reg_with_scaler = make_pipeline(
        StandardScaler(with_mean=with_mean),
        model,
    )
```

Reply:
making the name explicit is enough then